src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -292,7 +292,14 @@ private function getAutowiredReference(TypedReference $reference)
         }
 
         if (isset($this->types[$type])) {
-            @trigger_error(sprintf('Autowiring services based on the types they implement is deprecated since Symfony 3.3 and won\'t be supported in version 4.0. You should %s the "%s" service to "%s" instead.', isset($this->types[$this->types[$type]]) ? 'alias' : 'rename (or alias)', $this->types[$type], $type), E_USER_DEPRECATED);
+            $message = 'Autowiring services based on the types they implement is deprecated since Symfony 3.3 and won\'t be supported in version 4.0.';
+            if ($aliasSuggestion = $this->getAliasesSuggestionForType($type = $reference->getType())) {
+                $message .= ' '.$aliasSuggestion;
+            } else {
+                $message .= sprintf(' You should %s the "%s" service to "%s" instead.', isset($this->types[$this->types[$type]]) ? 'alias' : 'rename (or alias)', $this->types[$type], $type);
+            }
+
+            @trigger_error($message, E_USER_DEPRECATED);
```

Comment:
I am working on improving the message - you're right in another comment about there being no context (latest commit is at least partially there - you can see the before/after in the PR comment). I may need to tweak it a bit more still.

And this is a fair point. It's interesting... because telling them to alias (e.g. `EntityManager` -> `EntityManagerInterface`) is a really easy, global fix. But it's... sorta "wrong", since we're encouraging people to allow themselves to type to the class. But... if they're the ones doing it... is it such a big deal? If we think it's not such a big deal, then I think we should indeed revert to the old message.